mono/mini/mini-exceptions.c

```diff
@@ -2034,6 +2036,12 @@ mono_handle_exception_internal (MonoContext *ctx, MonoObject *obj, gboolean resu
 		} else {
 			free_stack = 0xffffff;
 		}
+
+		if (method->wrapper_type == MONO_WRAPPER_NATIVE_TO_MANAGED && ftnptr_eh_callback) {
+			guint32 handle = mono_gchandle_new (obj, FALSE);
+
```

Comment:
We have to call `mono_threads_detach_coop` here, otherwise the `ftnptr_eh_callback` will unwind in GC unsafe mode and that can lead to suspend locks (the runtime tries to suspend a thread but it's in GC unsafe mode and it doesn't hit any safe point)